File: 05- May/08- Matrix Diagonal Sum/08- Matrix Diagonal Sum (Noura Algohary).py
# Author: Noura Algohary
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n = len(mat)
        sm = 0
        for i in range(n):
            sm += mat[i][i]
            if i!=(n-1)/2:
                sm += mat[i][n-i-1]
            
        return sm
                        

File: 05- May/08- Matrix Diagonal Sum/08- Matrix Diagonal Sum (Noura Algohary).cpp
// Author: Noura Algohary
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int result = 0;
        int n = mat.size();

        for(int i = 0; i<n; i++)
        {
            for(int j = 0;j <n;j++)
                {
                    if(i + j == n - 1 || i == j)
                        result += mat[i][j];
                }
        }

        return result;
    }
};